src/lib.rs

```diff
@@ -198,67 +287,200 @@ impl IRCompilerFactory {
+        chs: u64,
+        miss: u64,
+        any_hit: u64,
+        callable_args: u64,
+        max_recursive_depth: i32,
+    ) {
+        unsafe {
+            self.funcs.IRCompilerSetRayTracingPipelineArguments(
+                self.me.as_ptr(),
+                max_attribute_size_in_bytes,
+                raytracing_pipeline_flags,
+                chs,
+                miss,
+                any_hit,
+                callable_args,
+                max_recursive_depth,
+            )
+        }
+    }
+
+    #[doc(alias = "IRCompilerSetHitgroupType")]
+    pub fn set_hitgroup_type(&mut self, hit_group_type: ffi::IRHitGroupType) {
+        unsafe {
+            self.funcs
+                .IRCompilerSetHitgroupType(self.me.as_ptr(), hit_group_type)
+        }
+    }
+
+    #[doc(alias = "IRMetalLibSynthesizeIndirectIntersectionFunction")]
+    pub fn synthesize_indirect_intersection_function(&mut self) -> Option<IRMetalLibBinary> {
+        let target_metallib = IRMetalLibBinary::new(self);
+        if unsafe {
+            self.funcs.IRMetalLibSynthesizeIndirectIntersectionFunction(
+                self.me.as_ptr(),
+                target_metallib.me.as_ptr(),
+            )
+        } {
+            Some(target_metallib)
+        } else {
+            None
+        }
+    }
+
+    #[doc(alias = "IRMetalLibSynthesizeIndirectRayDispatchFunction")]
+    pub fn synthesize_indirect_ray_dispatch_function(&mut self) -> Option<IRMetalLibBinary> {
+        let target_metallib = IRMetalLibBinary::new(self);
+        if unsafe {
+            self.funcs.IRMetalLibSynthesizeIndirectRayDispatchFunction(
+                self.me.as_ptr(),
+                target_metallib.me.as_ptr(),
+            )
+        } {
+            Some(target_metallib)
+        } else {
+            None
+        }
+    }
+
+    #[doc(alias = "IRCompilerSetEntryPointName")]
     pub fn set_entry_point_name(&mut self, new_name: &CStr) {
         unsafe {
             self.funcs
-                .IRCompilerSetEntryPointName(self.me, new_name.as_ptr())
+                .IRCompilerSetEntryPointName(self.me.as_ptr(), new_name.as_ptr())
         }
     }
 
+    #[doc(alias = "IRCompilerAllocCompileAndLink")]
     pub fn alloc_compile_and_link(
         &mut self,
         entry_point: &CStr,
         input: &IRObject,
-    ) -> Result<IRObject, Box<dyn std::error::Error>> {
+    ) -> Result<IRObject, CompilerError> {
         let mut error = std::ptr::null_mut();
 
-        let v = unsafe {
+        let v = NonNull::new(unsafe {
             self.funcs.IRCompilerAllocCompileAndLink(
-                self.me,
+                self.me.as_ptr(),
                 entry_point.as_ptr(),
-                input.me,
+                input.me.as_ptr(),
                 &mut error,
             )
-        };
+        });
 
         if error.is_null() {
             Ok(IRObject {
-                me: v,
+                me: v.unwrap(),
```

Comment:
Above you handled this with an `Unknown` error, but here we just unwrap?

What's the difference?